services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -626,6 +613,66 @@ def get_writer_batch_size(ds_config_info: datasets.info.DatasetInfo) -> Optional
         return None
 
 
+def copy_parquet_files(builder: DatasetBuilder) -> List[CommitOperationCopy]:
+    """Copy parquet files by copying the git LFS pointer files"""
+    data_files = builder.config.data_files
+    if not data_files:
+        raise EmptyDatasetError("Empty parquet data_files")
+    parquet_operations = []
+    for split in data_files:
+        num_shards = len(data_files[split])
```

Comment:
Ah yes indeed ! The limit is actually 10,000 since you can't have more than 10k files in the same directory.

I can raise an error in that case for now.

Reply:
Note that it's a (weak) argument for changing the structure to config/split/NNN.parquet (it would avoid putting all the splits in the same directory)